cardinal/component/manager.go

```diff
@@ -0,0 +1,82 @@
+package component
+
+import (
+	"fmt"
+	"github.com/rotisserie/eris"
+	"pkg.world.dev/world-engine/cardinal/storage/redis"
+	"pkg.world.dev/world-engine/cardinal/types"
+)
+
+var ErrComponentNotRegistered = eris.New("component not registered")
+
+type Manager struct {
+	registeredComponents map[string]types.ComponentMetadata
+	nextComponentID      types.ComponentID
+	storage              StorageProvider
+}
+
+// NewManager creates a new component manager.
+func NewManager(storage StorageProvider) *Manager {
+	return &Manager{
+		registeredComponents: make(map[string]types.ComponentMetadata),
+		nextComponentID:      1,
+		storage:              storage,
+	}
+}
+
+// RegisterComponent registers component with the component manager.
+// There can only be one component with a given name, which is declared by the user by implementing the Name() method.
+// If there is a duplicate component name, an error will be returned and the component will not be registered.
+func (m *Manager) RegisterComponent(compMetadata types.ComponentMetadata) error {
```

Comment:
It looks like you're missing the check to see if a component with the same name was already registered. The previous code was:

```
_, err := w.GetComponentByName(t.Name())
if err == nil {
    return eris.Errorf("component %q is already registered", t.Name())
}
```
There should be a test that complains about duplicate component names. If there isn't one, please add it.

Reply:
whoops, good catch

Reply:
fixed also added more tests